sdk/nodejs/cmd/pulumi-language-nodejs/testdata/tsc/projects/l2-component-call-simple/index.ts

```diff
@@ -2,7 +2,7 @@ import * as pulumi from "@pulumi/pulumi";
 import * as component from "@pulumi/component";
 
 const component1 = new component.ComponentCallable("component1", {value: "bar"});
-export const from_identity = component1.identity().result;
+export const from_identity = component1.identity().apply(call => call.result);
```

Comment:
Why aren't this all lifting? It would still work for python and TS to just be `identity().result`?

Reply:
If you mean "why are we generating code with `apply` when we don't need to?", I'm not sure. We don't appear to be consistent with it through NodeJS and Python programgen, but I've not had a chance to look into it yet.

Reply:
Yes that was the question. I'd guess where just not calling "lowerExpression" everywhere it's needed? But then why aren't we just doing a global lower all expressions before writing them out? Seems odd, don't think it needs to block this PR, but we should look into it.